refactor: use Command.Args.t for Modules.pp

emillon opened the pull request:
This allows using the concatenation DSL instead of list and string manipulation.

rgrinberg commented on src/dune_rules/command.mli:
```diff
@@ -63,6 +63,9 @@ module Args : sig
   val dyn : string list Action_builder.t -> _ t
 
+  (** Like [dyn] but output in a single argument, quoted and concatenated. *)
+  val dyn_as_single_arg : string list Action_builder.t -> _ t
```
Not a fan of this function. The name is weird and the function's semantics are best understood just by reading the implementation. It's also barely useful as it's only used once in the code base.

rgrinberg commented on src/dune_rules/module.mli:
```diff
@@ -61,7 +61,7 @@ val path : t -> Module_name.Path.t
 val source : t -> ml_kind:Ml_kind.t -> File.t option
 
-val pp_flags : t -> (string list Action_builder.t * Sandbox_config.t) option
+val pp_flags : t -> (Command.Args.any Command.Args.t * Sandbox_config.t) option
```
The old type was actually superior because it prevented the flags from introducing hidden targets. Command.Args.t doesn't do that. But even without that, the `string list Action_builder.t` is strictly less powerful and enough for this job. Why go for the bigger hammer?

emillon committed:
Use without_targets args

Signed-off-by: Etienne Millon <me@emillon.org>

emillon committed:
Remove changes in Module

Signed-off-by: Etienne Millon <me@emillon.org>

emillon commented:
Good points. The changes I wanted to do (use `Args` more) actually don't require changes to `Module` itself. How does that look now?

emillon committed:
refactor: use Command.Args.t for Modules.pp

This allows using the concatenation DSL instead of list and string
manipulation.

Signed-off-by: Etienne Millon <me@emillon.org>

emillon committed:
Merge branch 'main' into refactor-module-pp-args
